extensions/answer_summarizers/models.py

```diff
@@ -90,6 +96,52 @@ def _get_top_answers_by_frequency(answers, limit=None):
     ])
 
 
+def _get_top_unresolved_answers_by_frequency(
+        answers_with_classification, limit=None):
+    hashable_answers_with_classification = [{
+        'hashable_answer': _HashableAnswer(ans['answer']),
+        'classification_categorization': ans['classification_categorization']
+    } for ans in answers_with_classification]
+
+    classification_results_dict = {}
+    for ans in hashable_answers_with_classification:
+        frequency = 0
+        if ans['hashable_answer'] in classification_results_dict:
+            frequency = classification_results_dict[ans['hashable_answer']][
+                'frequency']
+        classification_results_dict[ans['hashable_answer']] = {
+            'classification_categorization': (
+                ans['classification_categorization']),
+            'frequency': frequency + 1
+        }
+
+    answers_with_most_recent_classification = [{
+        'answer': ans.answer,
+        'classification_categorization': val['classification_categorization'],
+        'frequency': val['frequency']
+    } for ans, val in classification_results_dict.iteritems()]
+
+    grouped_hashable_answers = itertools.groupby(
+        answers_with_most_recent_classification,
+        operator.itemgetter('classification_categorization'))
+
+    unresolved_answers_with_frequency_list = []
+    for category, answers in grouped_hashable_answers:
+        if category in UNRESOLVED_ANSWER_CLASSIFICATION_CATEGORIES:
```

Comment:
I feel like you can filter much earlier. Why not get rid of non-interesting answers in the very first for loop?

Reply:
Because we can not be sure whether an answer is resolved or unresolved until we have iterated through all of the answers.

For example an answer appearing earlier as resolved answer may appear as unresolved answer aterwards and similarly an answer appearing as unresolved may become resolved afterwards. This kind of changes can happen as answers are aggragated across multiple versions of exploration. That's why an answer can have different classification categorization in the list.

Reply:
I see, thanks for explaining. What about filtering in line 118? At that point you already have the most recent categorizations.

Reply:
On line 118, I am just restructuring the data so that it is a list (and not a dict) and we can use counter and groupby.

Reply:
Right -- but this involves a for loop over all the answers, and I suspect groupby() has nontrivial cost as well. I know it doesn't improve asymptotic complexity but when dealing with large amounts of data (as answers tend to be) speeding things up by a nontrivial fraction seems worth doing. 

Hence the suggestion to filter as early as possible so that you're only working with (hopefully) a much smaller quantity of unresolved answers, rather than everything from the entire answer submission history.

Reply:
Done